g11n-ws/tools/tool-cldr-extractor/src/main/java/com/vmware/i18n/utils/timezone/CldrTimeZoneUtils.java

```diff
@@ -9,142 +9,224 @@
-           
-            if (metazoneValue == null) {
-                TimeZoneDisplayName longPerp = new TimeZoneDisplayName();
-                String standard = MessageFormat.format(regionFormatTypeStandard, exemplarCity);
-                longPerp.setStandard(standard);
-                metaZonePerperties.put(Constants.LONG, longPerp);
-            } else {
-                String longStandard = (String) select(metazoneValue, "long.standard");
-                String longDaylight = (String) select(metazoneValue, "long.daylight");
-                String longGeneric = (String) select(metazoneValue, "long.generic");
-                String shortStandard = (String) select(metazoneValue, "short.standard");
-                String shortDaylight = (String) select(metazoneValue, "short.daylight");
-                String shortGeneric = (String) select(metazoneValue, "short.generic");
-                if(longStandard != null || longDaylight != null || longGeneric != null) {
-                	TimeZoneDisplayName longPerp = new TimeZoneDisplayName();
-                	longPerp.setDaylight(longDaylight);
-                	longPerp.setGeneric(longGeneric);
-                	longPerp.setStandard(longStandard);
-                	metaZonePerperties.put(Constants.LONG, longPerp);
-                }
-                if(shortStandard != null || shortDaylight != null || shortGeneric != null) {
-                	TimeZoneDisplayName shortPerp = new TimeZoneDisplayName();
-                	shortPerp.setDaylight(shortDaylight);
-                	shortPerp.setGeneric(shortGeneric);
-                	shortPerp.setStandard(shortStandard);
-                	metaZonePerperties.put(Constants.SHORT, shortPerp);
-                }
-               
-            }
-            metaZones.add(new CldrMetaZone(zoneKey, exemplarCity, metazoneKey, timeZone, metaZonePerperties,
-                    territory));
-        }
-
-        TimeZoneName zone = new TimeZoneName(language, gmtZeroFormat, gmtFormat, hourFormat,
-                regionFormat, regionFormatTypeDaylight, regionFormatTypeStandard, fallbackFormat,
-                metaZones);
-        String result = null;
-        try {
-            result = new ObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(zone);
-        } catch (JsonProcessingException e) {
-            // TODO Auto-generated catch block
-            e.printStackTrace();
-        }
-        return result;
-    }
 
-    private static String findTimeZone(String targetId, String gmtFormat, String gmtZeroFormat,
-            String hourFormat) {
+	@SuppressWarnings("unchecked")
+	public static Map<String, JSONArray> findTimezoneKeys(JSONObject metaZonesJson) {
+		JSONObject timzone = (JSONObject) select(metaZonesJson, "supplemental.metaZones.metazoneInfo.timezone");
+		Map<String, JSONArray> result = new TreeMap<String, JSONArray>();
+		for (Entry<String, Object> entry : (Set<Map.Entry<String, Object>>) timzone.entrySet()) {
```

Comment:
Most of the code here is the same, are there other ways to optimize it? Like iteration.

Reply:
this is the business required